lisa/analysis/tasks.py

```diff
@@ -423,6 +423,46 @@ def df_task_total_residency(self, task):
 
         return residency_df
 
+    @df_task_total_residency.used_events
+    def df_tasks_total_residency(self, tasks=None, ascending=False, count=None):
+        """
+        DataFrame of tasks execution time on each CPU
+
+        :param tasks: List of tasks to report, all trace tasks by default
+        :type tasks: list(int or str)
+
+        :param ascending: Set True to plot before short running tasks,
+                          False by deafult
+        :type ascending: bool
+
+        :param count: Maximum number of tasks to report
+        :type count: int
+        """
+        if tasks is None:
+            tasks = list(self.trace.get_tasks().keys())
+        res_df = pd.DataFrame()
+
+        # Ensure we have a list of PIDs
+        pids = []
+        for task in tasks:
```

Comment:
That whole loop should be
```python
pids = [self.trace.get_task_pid(task) for task in tasks]
```

Reply:
Are you sure that works in case `tasks` contains both PID numbers and task names?

Reply:
Yep:
```
Helper that takes either a name or a PID and always returns a PID
```